draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -765,16 +775,27 @@ responses to the address queries that were issued in parallel.
 A few initial SvcParamKeys are defined here.  These keys are useful for
 HTTPS, and most are applicable to other protocols as well.
 
-## "transport" and "no-default-transport"
+## "transport" and "no-default-transport" {#transport-key}
 
 The "transport" and "no-default-transport" SvcParamKeys together
 indicate the set of transport protocols supported by this service endpoint.
+A transport protocol is identified by a protocol-id with the following form:
+
+    protocol-id = 1*(ALPHA_LC / DIGIT / "-" / "_" / ".")
+
+The presentation and wire format of "transport" are the same: a comma (0x2c)
+separated list of one or more `protocol-id`s:
```

Comment:
Replace this with fixed-width length prefixes. A client parsing DNS (and TLS for that matter) already has tons of routines for such formats. Nothing the client parses in this spec may use separators.

Then revert the change to `protocol-id`. It should be an 8-bit-clean opaque byte string, like ALPN, so we do not have random implementation variations around people remembering to and forgetting to check, getting the check subtle wrong, and weird bugs in the corners.

Reply:
Makes sense to use fixed-width length prefixes to be consistent with everything else.

Reply:
Done.  I made them 1-byte prefix, so the max size is 255.